packages/serverpod/lib/src/server/server.dart

```diff
@@ -546,6 +546,7 @@ class Server {
     Endpoint endpoint,
   ) async {
     try {
+      session.sessionLogs.currentEndpoint = endpoint.name;
```

Comment:
Question: why is this necessary when closing the stream?

Reply:
The current endpoint is what is used for logging, so if some logging happens when we close the stream these logs will be reported for the wrong endpoint.